client/page.go

```diff
@@ -0,0 +1,133 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"reflect"
+	"regexp"
+)
+
+//Represents a page of records in a collection
+type Page struct {
+	Payload         map[string]interface{}
+	NextPageUrl     string
+	PreviousPageUrl string
+	Records         []interface{}
+	BaseUrl         string
+}
+
+func NewPage(baseUrl string, response interface{}) *Page {
+	var payload map[string]interface{}
+	if reflect.TypeOf(response).Kind() != reflect.Map {
+		test, err := json.Marshal(response)
+		if err != nil {
+			log.Print("Page creation error: ", err)
+			return nil
+		}
+		_ = json.Unmarshal(test, &payload)
+	} else {
+		payload = response.(map[string]interface{})
+	}
```

Comment:
Lots of conversions from response to interface and back again. Wondering if there's a different approach that we can take here using functions as arguments. Maybe try first to get it working where both list and stream utilize the specific type (maybe no hand-crafted code, all generated) and then try to refactor the common parts into hand-crafted pieces.